src/Query/Builder.php

```diff
@@ -279,6 +284,59 @@ public function dump(mixed ...$args)
         return $this;
     }
 
+    private function getAggregationBuilder(): AggregationBuilder
+    {
+        if (! trait_exists(FluentFactoryTrait::class)) {
+            throw new BadMethodCallException('Aggregation builder requires package mongodb/builder 0.2+');
+        }
+
+        $agg = new AggregationBuilder($this->collection, [], $this->options);
+
+        $wheres = $this->compileWheres();
+
+        if (count($wheres)) {
+            $agg->match(...$wheres);
+        }
+
+        // Distinct query
```

Comment:
Are any of these custom operations tested? Basically, anything that would result in `getAggregationBuilder()` prepending something to the pipeline should have a corresponding functional test.

I've having trouble grasping what the "distinct" case here does. It groups by a value (either `_id` or a column) and then reduces each grouped document to the first seen root document containing that value. That seems inconsistent with how the [`distinct` command](https://www.mongodb.com/docs/manual/reference/command/distinct/) works since it emits a list of the distinct values.

Reply:
It mimics the way DISTINCT works in SQL. I have added tests.